### PR TITLE
Culling fixes

### LINKED ISSUE #13036: [Undesired Feature] Full Preview Layout mode not showing the Overlay
**Background:**
Last night I was working on culling multiple (+150) after import. I normally use the Full Preview Layout mode since I know that the image in front of me is the image that will get start/rejection (I dont trust the culling layout with the mouse Act On, see https://github.com/darktable-org/darktable/issues/6025) plus it is large enough to show the details (closed eyes), not sharp focus.

**Issue:**
Since https://github.com/darktable-org/darktable/pull/12784, the overlay block only shows if I move the mouse to the top section and then goes away. This was the intent/Feature of this PR. On the first image, this is not a big problem because I normally was already using the mouse, but as soon I use the keyboard arrow to see the next image, the overlay doesnt show. It will only show if I move the mouse. I dont want carpal tunnel syndrome :-) 

While the overlay staying in the way is bad since it can block the image, not having any information is also bad. I cant see the image filename, I cant see if I rejected it (when trying to select the best one from a series), I cant tell if I hit the arrow twice and skipped an image via the filename. 

**To Reproduce**
- Press F 
- Press right keyboard arrow

**Expected behavior**
 I understand the desire for the overlay not staying ON all the time. Therefore I see these options:

- I would like the overlay to show up for some amount of time (configurable) and regardless of the mouse position. The overlay should show automatically for some amount of time if: 
  -  image is just loaded (eg. press arrow to see new image) 
  -  whenever a change is performed (eg. press Reject, Star). 

- Address https://github.com/darktable-org/darktable/issues/6025 to use culling layout (image is not as large at full preview, but ok for me via using W key). 

**Workaround**
- Last night I used FileManager Layout mode, set to display 1 image and turned on Permanent Overlays.
- Or start using Geeqie for culling. 

**Which commit introduced the issue**
https://github.com/darktable-org/darktable/pull/12784

**Platform**
* darktable version : 4.1.0~git1117 OSB Master
* OS                : Fedora 36 KDE X11

### LINKED ISSUE #13173: Scrolling leaves thumbnail overlay ghost image
**Describe the bug/issue**
Sometimes a ghost imprint of the thumbnail overlay can remain on screen when navigating through the file manager view.
image

**To Reproduce**
1. Enter filemanager layout
2. Place your mouse in an active element of the thumbnail overlay such as the stars or the rejection button.
3. Scroll your view via mouse wheel or via arrow keys.
4. A ghost image will remain on the previous thumbnail until you refresh the thumbnail by moving your mouse there again. (Even multiple ghosts can appear if you keep repeating the process).

**Platform**
* darktable version : list+4559~g6556a0785
* Linux - Distro    : Manjaro

### LINKED ISSUE #13182: Starting DT in Culling breaks timeline in Filemanager
**Describe the bug/issue**
When you start darktable while in culling and then switch to a filemanager layout, the timeline is not displayed.

**To Reproduce**
1. Enter Culling mode
2. Stop and start darktable
3. Go to Filemanager layout

--> The time line is not displayed correctly.

To fix, stay in file manager layout and restart darktable once more.

**Screenshots**
Broken
image

Working
image

**Which commit introduced the error**
Ran a bisect. Looks like
7cbd8537190b0e5593113805abbfacc6155db754
is to blame

### PR DESCRIPTION
**Pull request by AlicVB**
Some changes to fix multiple, recently reported, issues on culling and preview : 
- remove the "move mouse to bottom to hide overlay" feature in favor of a new shortcut ('i' by default) to show the overlay on demand if "no overlays" mode is set. This fixes #13036
- fix show of the overlay on image load (show right overlay and start timeout directly if needed)
- avoid "ghost" block overlays when using scroll or keyboard to move. This fixes #13173
- ensure the image under the cursor always has the focus, especially at culling startup
- fix the timeline visibility when starting in culling then going to filemanager. This fixes #13182

### CODE REVIEW

**Comment by gi-man** [comment]
> * remove the "move mouse to bottom to hide overlay" feature in favor of a new shortcut ('i' by default) to show the overlay on demand if "no overlays" mode is set. This fixes [[Undesired Feature] Full Preview Layout mode not showing the Overlay #13036](https://github.com/darktable-org/darktable/issues/13036)
> * fix show of the overlay on image load (show right overlay and start timeout directly if needed)
> * avoid "ghost" block overlays when using scroll or keyboard to move. This fixes [Scrolling leaves thumbnail overlay ghost image #13173](https://github.com/darktable-org/darktable/issues/13173)

I built and tested these and they worked for me. I liked the i shortcut idea.

File: src/libs/lib.c [hunk 1]
```diff
@@ -1106,7 +1106,7 @@ void dt_lib_set_visible(dt_lib_module_t *module, gboolean visible)
 
 
     if(visible)
-      gtk_widget_show(GTK_WIDGET(widget));
+      gtk_widget_show_all(GTK_WIDGET(widget));
```
[the remark this answers is not in the document]
Comment (AlicVB) [reply src/libs/lib.c]: yes, you may be right... I've force pushed a change in order to opt for a more local (timeline only) fix
Thanks for the (half-blind) review :)

File: src/dtgtk/culling.c [hunk 2]
```diff
@@ -1769,6 +1768,56 @@ void dt_culling_set_overlays_mode(dt_culling_t *table, dt_thumbnail_overlay_t ov
   g_free(cl1);
 }
 
+// force the overlays to be shown
+void dt_culling_force_overlay(dt_culling_t *table, gboolean force)
```
Comment (TurboGit) [inline src/dtgtk/culling.c:1772]: `const gboolean force`

File: src/dtgtk/culling.h [hunk 3]
```diff
@@ -95,6 +95,7 @@ void dt_culling_zoom_fit(dt_culling_t *table);
 
 // set the overlays type
 void dt_culling_set_overlays_mode(dt_culling_t *table, dt_thumbnail_overlay_t over);
+void dt_culling_force_overlay(dt_culling_t *table, gboolean force);
```
Comment (TurboGit) [inline src/dtgtk/culling.h:98]: `const gboolean force`

File: src/dtgtk/thumbnail.c [hunk 4]
```diff
@@ -1972,9 +1949,27 @@ static void _widget_change_parent_overlay(GtkWidget *w, GtkOverlay *new_parent)
 }
 void dt_thumbnail_set_overlay(dt_thumbnail_t *thumb, dt_thumbnail_overlay_t over, int timeout)
```
Comment (TurboGit) [inline src/dtgtk/thumbnail.c:1950]: `const int timeout`

File: src/dtgtk/thumbnail.h [hunk 5]
```diff
@@ -185,3 +184,4 @@ float dt_thumbnail_get_zoom_ratio(dt_thumbnail_t *thumb);
 // vim: shiftwidth=2 expandtab tabstop=2 cindent
 // kate: tab-indents: off; indent-width 2; replace-tabs on; indent-mode cstyle; remove-trailing-spaces modified;
 // clang-format on
+
```
Comment (TurboGit) [inline src/dtgtk/thumbnail.h:187]: superfluous

File: src/views/lighttable.c [hunk 6]
```diff
@@ -607,6 +607,29 @@ void scrollbar_changed(dt_view_t *self, double x, double y)
     dt_thumbtable_scrollbar_changed(dt_ui_thumbtable(darktable.gui->ui), x, y);
 }
 
+static void _overlays_force(dt_view_t *self, gboolean show)
```
Comment (TurboGit) [inline src/views/lighttable.c:610]: `const gboolean show`

File: src/views/lighttable.c [hunk 7]
```diff
@@ -652,6 +675,33 @@ const dt_action_def_t dt_action_def_preview
       _action_elements_preview,
       NULL, TRUE };
 
+static float _action_process_infos(gpointer target, dt_action_element_t element, dt_action_effect_t effect,
+                                   float move_size)
```
Comment (TurboGit) [inline src/views/lighttable.c:679]: `const float move_size`

File: src/views/lighttable.c [hunk 8]
```diff
@@ -652,6 +675,33 @@ const dt_action_def_t dt_action_def_preview
       _action_elements_preview,
       NULL, TRUE };
 
+static float _action_process_infos(gpointer target, dt_action_element_t element, dt_action_effect_t effect,
```
Comment (TurboGit) [inline src/views/lighttable.c:678]: `const dt_action_effect_t effect`

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some minor comments.

One question though, the key i make it possible to display info if otherwise hidden. But then how to hide them?

**Comment by dterrahe** [comment]
Still only on phone, so not able to check exactly what you are doing, but normally a _hold_ shortcut is meant to set a state variable that applies as long as it is pressed or between toggles. Here there seems to be a one time action when you press the shortcut, but if then you move to the next image, does it still apply?

**Comment by TurboGit** [comment]
> Here there seems to be a one time action when you press the shortcut, but if then you move to the next image, does it still apply?

When I press the info is displayed as expected (if it has been removed after timeout). But each time I move from one image to another I still have the info displayed. Looking at the comment it reads as if one can hide those info and have them displayed only when hitting `i`... But I cannot make this to work... @AlicVB ?

File: src/views/lighttable.c (hunk 7 above)
Comment (AlicVB) [reply src/views/lighttable.c]: I've fixed that, and the other `const` in the same function.
That said, @dterrahe : there's a lot of other place whith same function, where the const is not here... Is there a reason for that ? can I 'fix' them too ?

**Comment by AlicVB** [comment]
@TurboGit : in fact the purpose of the `i` shortcut is to show the block overlay during the key press, like the `h` does...
Following previous discussion and remarks, the idea, from an user POV is : 
1. I need the infos constantly for each image => overlay "permanent" OR overlay "block" with timeout `-1`
2. I need the info of each image "briefly" => overlay "block" with timeout >=0
3. I never or rarely need the infos => overlay "hidden"

For point 2 and 3, now, if you need temporarily the infos for some reason, you have the `i` shortcut to show the info on demand...

So iirc, for your usecase, you may want to use the overlay "hidden" and press `i` when you need

File: src/views/lighttable.c (hunk 7 above)
Comment (TurboGit) [reply src/views/lighttable.c]: > I've fixed that, and the other `const` in the same function. That said, @dterrahe : there's a lot of other place whith same function, where the const is not here... Is there a reason for that ? can I 'fix' them too ?

There is probably no good reasons. The const is often forgotten as not mandatory.  But I encourage all devs to fix them when they encounter some.

**Review by TurboGit — COMMENTED** [review]
Thanks, merged in 4.2.

**Comment by cyberspeck** [comment]
How can achieve suggested point 3? I couldn't find a setting to make the overlays 'hidden' in full preview mode. Thanks.

**Comment by quovadit** [comment]
> How can achieve suggested point 3? I couldn't find a setting to make the overlays 'hidden' in full preview mode. Thanks.

I would also like to disable the overlay in full preview layout. To show it there now is the 'i' shortcut. But it always is displayed everytime I move  to the next image using the arrow-keys.

**Comment by AlicVB** [comment]
That's a little bit tricky, as the overlays ("star") icon present in normal lighttable is hidden in full preview.
What you can when you are in full preview mode is to press "b" in order to show borders arrows. Then you can show the top panel using the top one... and set the overlays to none.
Once that done, just click again on the arrow to hide the top panel and press b again to hide the border arrows.

I admit that it's not really simple. Maybe at some point someone will have time to propose and code a better UI ?

**Comment by cyberspeck** [comment]
Thank you!!

**Pull request merged**